### PR TITLE
Wired grant request page into service flow

### PR DESCRIPTION
**Pull request by stuartmccoll**
This pull request wires the 'Your grant request' page into the service journey.

### CODE REVIEW

File: app/controllers/project/project_grant_request_controller.rb [hunk 1]
```diff
@@ -1,12 +1,63 @@
...
+      logger.debug "Retrieved project cost ID #{project_cost.id}, amount #{project_cost.amount} " +
+                       "for project ID: #{@project.id}"
+
+      @total_project_cost += project_cost.amount
+
+      logger.debug "Running total project cost for project ID: #{@project.id} is: #{@total_project_cost}"
+
+    end
+
+    logger.debug "Calculated total project cost of #{@total_project_cost} for project ID: #{@project.id}"
```
Comment (gidsg) [inline app/controllers/project/project_grant_request_controller.rb]: I think we could just use this existing helper

File: app/controllers/project/project_grant_request_controller.rb [hunk 2]
```diff
@@ -1,12 +1,63 @@
...
+                       "for project ID: #{@project.id}"
+
+      @total_cash_contributions += cash_contribution.amount
+
+      logger.debug "Running total cash contributions for project ID: #{@project.id} is: #{@total_cash_contributions}"
+
+    end
+
+    logger.debug "Calculated total cash contributions of #{@total_cash_contributions} for project ID: #{@project.id}"
+
```
Comment (gidsg) [inline app/controllers/project/project_grant_request_controller.rb]: I think we could just use or adapt this existing helper: https://github.com/heritagefund/funding-frontend/blob/650c5ab44f9a6ed12c3940a91671c2cc5df87092/app/helpers/project/project_costs_helper.rb#L2-L4

File: app/views/project/project_grant_request/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,78 @@
+<% content_for :page_title, "Your grant request" %>
+
+<h1 class="govuk-heading-xl">
+  <span class="govuk-caption-xl">About your project</span>
+  Your grant request
+</h1>
+
+<div class="govuk-panel govuk-panel--confirmation govuk-!-margin-bottom-7">
+  <h1 class="govuk-panel__title">
+    £<%= '%.2f' % @final_grant_amount %>
```
Comment (gidsg) [inline app/views/project/project_grant_request/show.html.erb]: We can use a helper which includes the pound symbol and formatting the number e.g.
```suggestion
    <%= number_to_currency(@final_grant_amount, strip_insignificant_zeros: true) %>
```

File: app/views/project/project_grant_request/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+      <h2 class="govuk-heading-s govuk-!-margin-bottom-3">Your cash contributions</h2>
+
+      <p class="govuk-body">This is the amount of money you told us you will receive for your project that does not include your grant request</p>
+
+    </div>
+
+    <div class="nlhf-grant-summary__amount">
+
+      <p>£<%= @total_cash_contributions %></p>
```
Comment (gidsg) [inline app/views/project/project_grant_request/show.html.erb]:
```suggestion
      <p><%= number_to_currency(@total_cash_contributions, strip_insignificant_zeros: true) %></p>
```

File: app/controllers/project/project_grant_request_controller.rb [hunk 5]
```diff
@@ -1,12 +1,63 @@
...
-    @total_project_cost = 100
-    @cash_contributions = 50
-    @final_grant_amount = @total_project_cost - @cash_contributions
+  include ProjectContext
+
+  def show
+
+    calculate_total_project_cost
+
+    calculate_total_cash_contributions
```
Comment (gidsg) [inline app/controllers/project/project_grant_request_controller.rb]: Can we remove these, as they are not assigned I don't _think_ they are doing anything?

Comment (stuartmccoll) [reply app/controllers/project/project_grant_request_controller.rb]: They're calling the two `private` methods, but I'll look at adapting the existing helper.

File: app/controllers/project/project_grant_request_controller.rb [hunk 6]
```diff
@@ -3,61 +3,11 @@ class Project::ProjectGrantRequestController < ApplicationController
 
   def show
 
-    calculate_total_project_cost
-
-    calculate_total_cash_contributions
+    @total_project_cost = helpers.calculate_total(@project.project_costs)
+    @total_cash_contributions = helpers.calculate_total(@project.cash_contributions)
```
Comment (gidsg) [inline app/controllers/project/project_grant_request_controller.rb]: I don't _think_ these need to be instance variables as we don't appear to be using them in the view? Only to calculate the value below?

Comment (stuartmccoll) [reply app/controllers/project/project_grant_request_controller.rb]: They get used in the 'Here's how we calculated this' section.

File: app/views/project/project_grant_request/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+      <h2 class="govuk-heading-s govuk-!-margin-bottom-3">Your total project costs</h2>
+
+      <p class="govuk-body">This is what you told us the total amount of money your whole project will cost</p>
+
+    </div>
+
+    <div class="nlhf-grant-summary__amount">
+
+      <p>£<%= @total_project_cost %></p>
```
Comment (gidsg) [inline app/views/project/project_grant_request/show.html.erb]: we can use the currency helper here.
```suggestion
      <p><%= number_to_currency(@total_project_cost, strip_insignificant_zeros: true) %></p>
```

File: app/views/project/project_grant_request/show.html.erb [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+<h1 class="govuk-heading-xl">
+  <span class="govuk-caption-xl">About your project</span>
+  Your grant request
+</h1>
+
+<div class="govuk-panel govuk-panel--confirmation govuk-!-margin-bottom-7">
+  <h1 class="govuk-panel__title">
+    <%= number_to_currency(@final_grant_amount, strip_insignificant_zeros: true) %>
+  </h1>
```
Comment (gidsg) [inline app/views/project/project_grant_request/show.html.erb]: we should not have multiple H1s on the page. Maybe make these both H2s?

**Pull request merged**